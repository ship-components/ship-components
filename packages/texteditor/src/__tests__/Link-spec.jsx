/**
 * Test Link component
 */

import React from 'react';
import TestUtils from 'react-dom/test-utils';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Link', () => {
  const Link = require('../Link').default;

  it('should render without error', () => {
    let element = React.createElement(
       Link,
       {decoratedText: 'www.internet.com'},
       'test'
    );

    expect(() => TestUtils.renderIntoDocument(element))
       .not.toThrow();
  });

  it('should match snapshot', () => {
    const wrapper = mount(
      <Link
        alt='100 years forever Rick and Morty'
        title='Interdimensional Twitter'
        decoratedText='https://twitter.com/RickandMorty'
        className='real-classy-stuff'
      >
        Wubba lubba dub dub!
      </Link>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
