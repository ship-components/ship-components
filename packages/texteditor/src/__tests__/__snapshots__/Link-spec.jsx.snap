// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should match snapshot 1`] = `
<Link
  alt="100 years forever Rick and Morty"
  className="real-classy-stuff"
  decoratedText="https://twitter.com/RickandMorty"
  target="_blank"
  title="Interdimensional Twitter"
>
  <a
    alt="100 years forever Rick and Morty"
    className="real-classy-stuff"
    href="https://twitter.com/RickandMorty"
    target="_blank"
    title="Interdimensional Twitter"
  >
    Wubba lubba dub dub!
  </a>
</Link>
`;
